File: edutech-backend/models/feedback.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class Feedback(BaseModel):
    user_id: int
    rating: int  # 1-5 stars
    category: str  # "general", "quiz", "chatbot", "summarization", "ui"
    subject: Optional[str] = None
    message: str
    submitted_at: datetime
    is_resolved: bool = False

class FeedbackResponse(BaseModel):
    id: int
    user_id: int
    rating: int
    category: str
    subject: Optional[str] = None
    message: str
    submitted_at: datetime
    is_resolved: bool

class FeedbackStats(BaseModel):
    total_feedbacks: int
    average_rating: float
    category_breakdown: dict
    recent_feedbacks: int

File: edutech-frontend/types/feedback.ts
export interface FeedbackSubmission {
  rating: number;
  category: string;
  message: string;
  subject?: string;
}

export interface FeedbackResponse {
  id: string;
  user_id: string;
  rating: number;
  category: string;
  subject?: string;
  message: string;
  submitted_at: string;
  is_resolved: boolean;
}

export interface FeedbackStats {
  total_feedbacks: number;
  average_rating: number;
  category_breakdown: Record<string, number>;
  recent_feedbacks: number;
}